packages/typechain/test/cli/parseArgs.test.ts

```diff
@@ -62,4 +62,36 @@ describe('cli > parseArgs', () => {
       flags: { alwaysGenerateOverloads: true },
     })
   })
+
+  it('shows usage guide given --help flag', () => {
+    process.argv = ['', '', '--help']
+
+    const processExit = process.exit
+    // eslint-disable-next-line no-console
+    const consoleLog = console.log
+
+    const exitMock = mockFn<typeof process.exit>(() => {
+      throw new Error('process.exit called')
+    })
+    process.exit = exitMock
+    const logMock = mockFn<typeof console.log>(() => {})
+    // eslint-disable-next-line no-console
+    console.log = logMock
+
+    try {
+      parseArgs()
+    } catch (err) {
+      expect((err as Error).message).toEqual('process.exit called')
+    }
+
+    expect(exitMock).toHaveBeenCalledWith([0])
+    const logged = logMock.calls[0].args[0] as string
+    expect(logged.includes('Options')).toEqual(true)
```

Comment:
probably a manner of taste but I would personally use dedicated matchers for string ie.

```suggestion
    expect(logged).toEqual(expect.stringMatching('Options'))
```

Sidenote: we should add a validator in earl for this too ie: 

```
expect(logged).toBeAStringMatching('Options')
```
Once we merge docs generation i will for on this 😆